### PR TITLE
Asan

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

**Comment by nirvn** [comment]
👀

File: vcpkg/triplets/arm-android.cmake [hunk 1]
```diff
@@ -5,5 +5,9 @@ set(VCPKG_CMAKE_SYSTEM_NAME Android)
 set(VCPKG_BUILD_TYPE release)
 
 set(ENV{VCPKG_ANDROID_NATIVE_API_LEVEL} "detect")
-set(ENV{CXXFLAGS} "-fstack-protector-strong")
-set(ENV{CFLAGS} "-fstack-protector-strong")
+if(PORT STREQUAL "proj4" OR PORT STREQUAL "proj")
+set(ENV{CXXFLAGS} "-fstack-protector-strong -fsanitize=address -fno-omit-frame-pointer")
+set(ENV{CFLAGS} "-fstack-protector-strong -fsanitize=address -fno-omit-frame-pointer")
+set(ENV{LFLAGS} "-fsanitize=address")
```
Comment (m-kuhn) [inline vcpkg/triplets/arm-android.cmake:11]:
```suggestion
set(ENV{LDFLAGS} "-fsanitize=address")

```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 76fe3405f18e72c33d95fc992c2e7062d1a05324:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-b0c6d2339e2ad465fe6f95091c07da55b7780a79-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-b0c6d2339e2ad465fe6f95091c07da55b7780a79-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-b0c6d2339e2ad465fe6f95091c07da55b7780a79-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-b0c6d2339e2ad465fe6f95091c07da55b7780a79-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 4e5293ef2db2245b95ebebb9f6035cffe4a8ab08:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-96f92e1de78e6f64e30bf7976d675dbedea2b1a6-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-96f92e1de78e6f64e30bf7976d675dbedea2b1a6-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-96f92e1de78e6f64e30bf7976d675dbedea2b1a6-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-96f92e1de78e6f64e30bf7976d675dbedea2b1a6-x86-android.apk)

**Comment by m-kuhn** [comment]
💥

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f74cab5cfb5461f25262b57e8eb15005d6f8989c:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-e94ce49023d218e265c3dc8dc0447744fd7cf090-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-e94ce49023d218e265c3dc8dc0447744fd7cf090-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-e94ce49023d218e265c3dc8dc0447744fd7cf090-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-e94ce49023d218e265c3dc8dc0447744fd7cf090-x86-android.apk)

**Comment by nirvn** [comment]
Tried the last available APK,  and also got a 💥. I'll provide a logcat later this morning.

**Comment by nirvn** [comment]
logcat:

```
=================================================================
==27955==ERROR: AddressSanitizer: SEGV on unknown address 0x1680001faf9290c0 (pc 0x007cf534aae0 bp 0x007d15d70e30 sp 0x007d15d70d60 T0)
==27955==The signal is caused by a READ memory access.
    #0 0x7cf534aadc  (/data/app/~~Ruidq0LVzyLDIzmcyLduXQ==/ch.opengis.qfield_beta-k-uJqulT7nMEtOGXbRELvw==/lib/arm64/libqfield_arm64-v8a.so+0x949adc)
    #1 0x7e4c8f7ac0  (/system/bin/app_process64+0x52ac0)
    #2 0x7e4c8e1f3c  (/system/bin/app_process64+0x3cf3c)
    #3 0x7e4c8dd0fc  (/system/bin/app_process64+0x380fc)
    #4 0x7e208f610c  (/apex/com.android.runtime/lib64/bionic/libdl.so+0x110c)
    #5 0x7e2c72edcc  (/apex/com.android.art/lib64/libnativeloader.so+0x1bdcc)
    #6 0x7e2c71fe38  (/apex/com.android.art/lib64/libnativeloader.so+0xce38)
    #7 0x7e2c71f568  (/apex/com.android.art/lib64/libnativeloader.so+0xc568)
    #8 0x7d97e6ddac  (/apex/com.android.art/lib64/libart.so+0x46ddac)
    #9 0x7d8836e0dc  (/apex/com.android.art/lib64/libopenjdkjvm.so+0x50dc)
    #10 0x6f3e1834  (/apex/com.android.art/javalib/arm64/boot.oat+0xab834)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (/data/app/~~Ruidq0LVzyLDIzmcyLduXQ==/ch.opengis.qfield_beta-k-uJqulT7nMEtOGXbRELvw==/lib/arm64/libqfield_arm64-v8a.so+0x949adc) 
==27955==ABORTING
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 0b6f908effe7c6cdd728fef16e869058afecde1b:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-3d246e9c057c671c72bdeff5e43db183712af7aa-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-3d246e9c057c671c72bdeff5e43db183712af7aa-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-3d246e9c057c671c72bdeff5e43db183712af7aa-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-3d246e9c057c671c72bdeff5e43db183712af7aa-x86-android.apk)

**Comment by nirvn** [comment]
Same result for latest builds I produced this morning.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 8c34939f002e5fb7cbc5a2e85b6d76eb8becd8f4:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-586d3b78abf7f90b18aceab75a5e76bd484c9bb8-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-586d3b78abf7f90b18aceab75a5e76bd484c9bb8-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-586d3b78abf7f90b18aceab75a5e76bd484c9bb8-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-586d3b78abf7f90b18aceab75a5e76bd484c9bb8-x86-android.apk)

File: vcpkg/triplets/arm-android.cmake [hunk 2]
```diff
@@ -4,6 +4,18 @@ set(VCPKG_LIBRARY_LINKAGE static)
...
-set(ENV{CXXFLAGS} "-fstack-protector-strong")
-set(ENV{CFLAGS} "-fstack-protector-strong")
+set(ENV{CXXFLAGS} "-fstack-protector-strong -fsanitize=address -fno-omit-frame-pointer")
+set(ENV{CXXFLAGS_RELEASE} "-fstack-protector-strong -fsanitize=address -fno-omit-frame-pointer")
+set(ENV{CFLAGS} "-fstack-protector-strong -fsanitize=address -fno-omit-frame-pointer")
+set(ENV{CFLAGS_RELEASE} "-fstack-protector-strong -fsanitize=address -fno-omit-frame-pointer")
+set(ENV{LDFLAGS} "-fsanitize=address")
+set(ENV{LDFLAGS_RELEASE} "-fsanitize=address")
+
+set(VCPKG_CXX_FLAGS "-fstack-protector-strong -fsanitize=address -fno-omit-frame-pointer")
```
Comment (m-kuhn) [inline vcpkg/triplets/arm-android.cmake:14]: I think we need:

VCPKG_CXX_FLAGS, VCPKG_C_FLAGS, VCPKG_LINKER_FLAGS and could drop the rest.

Comment (nirvn) [reply vcpkg/triplets/arm-android.cmake:7]: @m-kuhn , done.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c2545c8ee34ccd4d507114801ebcf25f0bf55a46:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-3b6271dd9457f7d81f394d609d8d6f2b623708df-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-3b6271dd9457f7d81f394d609d8d6f2b623708df-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-3b6271dd9457f7d81f394d609d8d6f2b623708df-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-3b6271dd9457f7d81f394d609d8d6f2b623708df-x86-android.apk)

**Comment by nirvn** [comment]
Details on arm64 startup crash:
```
=================================================================
==26680==ERROR: AddressSanitizer: SEGV on unknown address 0x1680001fb1bdaa88 (pc 0x007ce6a1d8ec bp 0x007d7ebc5b60 sp 0x007d7ebc5ac0 T0)
==26680==The signal is caused by a READ memory access.
    #0 0x7ce6a1d8ec  (/data/app/~~xX7MUXfbVzianN9CXSzpkw==/ch.opengis.qfield_beta-Xv7wp13_5HMnqJEo5N7ogg==/lib/arm64/libqfield_arm64-v8a.so+0x7db08ec)
    #1 0x7ce3b1d3a0  (/data/app/~~xX7MUXfbVzianN9CXSzpkw==/ch.opengis.qfield_beta-Xv7wp13_5HMnqJEo5N7ogg==/lib/arm64/libqfield_arm64-v8a.so+0x4eb03a0)
    #2 0x7e4c8f7ac0  (/system/bin/app_process64+0x52ac0)
    #3 0x7e4c8e1f3c  (/system/bin/app_process64+0x3cf3c)
    #4 0x7e4c8dd0fc  (/system/bin/app_process64+0x380fc)
    #5 0x7e208f610c  (/apex/com.android.runtime/lib64/bionic/libdl.so+0x110c)
    #6 0x7e2c72edcc  (/apex/com.android.art/lib64/libnativeloader.so+0x1bdcc)
    #7 0x7e2c71fe38  (/apex/com.android.art/lib64/libnativeloader.so+0xce38)
    #8 0x7e2c71f568  (/apex/com.android.art/lib64/libnativeloader.so+0xc568)
    #9 0x7d97e6ddac  (/apex/com.android.art/lib64/libart.so+0x46ddac)
    #10 0x7d8836e0dc  (/apex/com.android.art/lib64/libopenjdkjvm.so+0x50dc)
    #11 0x6f3e1834  (/apex/com.android.art/javalib/arm64/boot.oat+0xab834)

AddressSanitizer can not provide additional info.
```

The first two addresses are referring to the following lines:
```
#1 int std::__ndk1::__cxx_atomic_load<int>(std::__ndk1::__cxx_atomic_base_impl<int> const*, std::__ndk1::memory_order)
/usr/local/lib/android/sdk/ndk/22.1.7171670/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/atomic:977

#2 __cxx_global_var_init.5
/home/runner/builddir/vcpkg_installed/arm64-android/include/qgis/qgsnetworkaccessmanager.h:612
```

The QGIS source code line is (https://github.com/qgis/QGIS/blob/release-3_24/src/core/network/qgsnetworkaccessmanager.h#L612): `static const inline QgsSettingsEntryInteger settingsNetworkTimeout = QgsSettingsEntryInteger( QStringLiteral( "/qgis/networkAndProxy/networkTimeout" ), QgsSettings::NoSection, 60000, QObject::tr( "Network timeout" ) );`

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for b03eb9d585a1c927ae93fdd68a579b2fe489f690:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-c9743d8fdd6137086ef1293b802d0167d6e6689b-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-c9743d8fdd6137086ef1293b802d0167d6e6689b-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-c9743d8fdd6137086ef1293b802d0167d6e6689b-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-c9743d8fdd6137086ef1293b802d0167d6e6689b-x86-android.apk)

**Comment by nirvn** [comment]
@marioba 's phone crash logcat using these builds:

```
=================================================================
==25878==ERROR: AddressSanitizer: SEGV on unknown address 0x0000000a (pc 0xb09b1096 bp 0xa755f400 sp 0xa755f3b0 T0)
==25878==The signal is caused by a READ memory access.
==25878==Hint: address points to the zero page.
```

There's unfortunately nothing more than that -- there's not even an android stacktrace in there, just these three lines of info.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a804547d425fb5d95f533eba490470d4cbd4077d:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-ebe5f042b21bc44cb0081cfccc03a12ae446ae68-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-ebe5f042b21bc44cb0081cfccc03a12ae446ae68-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-ebe5f042b21bc44cb0081cfccc03a12ae446ae68-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-ebe5f042b21bc44cb0081cfccc03a12ae446ae68-x86-android.apk)

File: vcpkg/overlay/qgis/portfile.cmake [hunk 3]
```diff
@@ -1,5 +1,5 @@
-set(QGIS_REF 78789804b5edf89777fe06b3fbeb88a8ef8f05e2)
-set(QGIS_SHA512 3035289511ea864e667930d9d17c38e50aed9f72e6065804ff01ca5a5de91b0be15be7238d1354241600578ce901373a09804eb50dadcfc0a0727df32f9a42bf)
+set(QGIS_REF 650a203e269aaa285dde8e37623a0c5129f094b1)
```
Comment (m-kuhn) [inline vcpkg/overlay/qgis/portfile.cmake:1]: Is there a tag name (for readability)?

Comment (nirvn) [reply vcpkg/overlay/qgis/portfile.cmake:1]: That's current master. Not planning to settle on this revision, merely wanted to see if the settings asan error was still there.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 81bf21b4bdb09b278a6b2e9dbaf312a1f4119658:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-e050431e5c8cadd7a58d3881f61f302ef4e1fa32-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-e050431e5c8cadd7a58d3881f61f302ef4e1fa32-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-e050431e5c8cadd7a58d3881f61f302ef4e1fa32-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2710-e050431e5c8cadd7a58d3881f61f302ef4e1fa32-x86-android.apk)

**Pull request closed without merging**